usaspending_api/common/views.py

```diff
@@ -105,29 +43,33 @@ class DetailViewSet(viewsets.ReadOnlyModelViewSet):
     view of model objects (either in the form of a single
     object or a list of objects).
     """
-    # Note: once the front-end has switched to using query parameter
-    # pagination for endpoints that use DetailViewSet, we can remove
-    # this class and inherit from ReadOnlyModelViewSet directly in
-    # the application views.py files.
 
     exception_logger = logging.getLogger("exceptions")
 
     @cache_response()
     def list(self, request, *args, **kwargs):
         try:
+            # First, get the req object from RequestCatalog
             created, self.req = RequestCatalog.get_or_create_from_request(request)
-            response = self.build_response(
-                request, queryset=self.get_queryset(), serializer=self.get_serializer_class())
-            status_code = status.HTTP_200_OK
+            # Pass this to the paginator
+            self.paginator.req = self.req
+            # Get the queryset (the view will handle filtering and ordering)
```

Comment:
I'll update to clarify, it means the view for the app, not the viewset here (i.e. get_queryset handles it)